qutip/tests/core/data/test_csr.py

```diff
@@ -343,6 +343,105 @@ def test_one_element_error(self, shape, position, value):
                                          " out of bound: ")
 
 
+class TestFromCSRBlocks:
+    class _blocks:
+        def __init__(self, rows, cols, ops, n_blocks=2, block_size=2):
+            self.rows = np.array(rows, dtype=data.base.idxint_dtype)
+            self.cols = np.array(cols, dtype=data.base.idxint_dtype)
+            if isinstance(ops, int):
+                ops = [csr.empty(block_size, block_size, 0)] * ops
```

Comment:
```suggestion
            if isinstance(ops, int):
                ops = [csr.zeros(block_size, block_size, 0)] * ops
```
Here I feels `csr.zeros` is what you want. 
`csr.zeros` is a sparse array with no elements.
`csr.empty`, like `np.empty`, allocate the memory for you to fill manually, but it's not to use as is.

Reply:
Thanks! I realized that `csr.empty` is really an internal method half way through and then forgot to clean up all the cases in the tests and outputs. I have sorted them out now.